src/backend/aspen/api/utils/fasta_streamer.py

```diff
@@ -21,28 +21,31 @@ class FastaStreamer:
     def __init__(
         self,
         db: AsyncSession,
+        az: AuthZSession,
         user: User,
         sample_ids: Set[str],
         downstream_consumer: Optional[str] = None,
     ):
         self.db = db
         self.user = user
+        self.az = az
+        self.sample_ids = sample_ids
+        # Certain consumers have different requirements on fasta
+        self.downstream_consumer = downstream_consumer
+
+    async def stream(self) -> AsyncGenerator[str, None]:
         # query for samples
-        all_samples_query = sa.select(Sample).options(  # type: ignore
+        sample_query = await samples_by_identifiers(
+            self.az, self.sample_ids, "sequences"
+        )
+        sample_query = sample_query.options(  # type: ignore
```

Comment:
Just moved this query into stream() since we can't await in constructors. Shouldn't really work any differently.